src/yb/master/backfill_index.cc

```diff
@@ -678,6 +667,27 @@ BackfillTable::BackfillTable(
   done_.store(false, std::memory_order_release);
 }
 
+std::string BackfillTable::RetrieveIndexNames(const std::unordered_set<std::string>& index_ids) {
```

Comment:
@jaki I don't know if this is workable, mainly because when trying to call ```Master::catalog__manager()``` in ```backfill_index.h```, the following error was obtained:
```
In file included from src/yb/master/backfill_index.cc:13:0:
src/yb/master/backfill_index.h: In constructor 'yb::master::BackfillChunk::BackfillChunk(std::shared_ptr<yb::master::BackfillTablet>, const string&)':
src/yb/master/backfill_index.h:345:57: error: invalid use of incomplete type 'class yb::master::Master'
         requested_index_names_(backfill_tablet->master()->catalog_manager()
                                                         ^
In file included from ent/src/yb/master/async_rpc_tasks.h:16:0,
                 from src/yb/master/backfill_index.h:22,
                 from src/yb/master/backfill_index.cc:13:
src/yb/master/async_rpc_tasks.h:59:7: note: forward declaration of 'class yb::master::Master'
 class Master;
       ^
```
I can call the ```catalog_manager.cc``` class in ```backfill_index.cc``` method, not in the .h file. The main problem is that the statement where the error is thrown is in BackfillChunk which is located in .h

Reply:
@ConcurrencyPractitioner, okay.  How about anonymous namespace, like for `IndexIdsFromInfos`?

Reply:
Actually...try moving the constructor definition to the .cc, then.  Both suggestions don't work without that.

Reply:
@jaki Cool, sounds good. I've made the change.